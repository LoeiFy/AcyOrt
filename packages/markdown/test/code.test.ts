import Markdown from '../src'

it('code', () => {
  const m0 = new Markdown()

  const c0 = `\`\`\`html
  <h1>h1</h1>
\`\`\``
  expect(m0.render(c0)).toMatchSnapshot()
  expect(m0.render(c0, { lineNumbers: false })).toMatchSnapshot()
  expect(m0.render(c0, { lineNumbers: true })).toMatchSnapshot()

  const c1 = `\`\`\`javascript
  var a = 1;
\`\`\``
  expect(m0.render(c1)).toMatchSnapshot()

  const c2 = `\`\`\`
  <h1>h1</h1>
\`\`\``
  expect(m0.render(c2)).toMatchSnapshot()
  expect(m0.render(c2, { lineNumbers: true })).toMatchSnapshot()

  const c3 = `\`\`\`sss
  <h1>h1</h1>
\`\`\``
  expect(m0.render(c3)).toMatchSnapshot()

  const m1 = new Markdown({ lineNumbers: true })
  expect(m1.render(c0)).toMatchSnapshot()

  const m2 = new Markdown({ lineNumbers: false })
  expect(m2.render(c0)).toMatchSnapshot()
})
