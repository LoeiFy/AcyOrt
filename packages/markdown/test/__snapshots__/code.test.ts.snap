// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`code 1`] = `
"<pre class="language-html">
  <code class="language-html">  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>h1</span><span class="token punctuation">></span></span>h1<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>h1</span><span class="token punctuation">></span></span></code>
</pre>"
`;

exports[`code 2`] = `
"<pre class="language-html">
  <code class="language-html">  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>h1</span><span class="token punctuation">></span></span>h1<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>h1</span><span class="token punctuation">></span></span></code>
</pre>"
`;

exports[`code 3`] = `
"<div class="code-highlight">
  <table>
    <tbody>
      <tr>
        <td class="code-highlight-line"><pre class="language-none">
  <code class="language-none">
    <span>1</span>

  </code>
</pre></td>
        <td class="code-highlight-code"><pre class="language-html">
  <code class="language-html">  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>h1</span><span class="token punctuation">></span></span>h1<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>h1</span><span class="token punctuation">></span></span></code>
</pre></td>
      </tr>
    </tbody>
  </table>
</div>"
`;

exports[`code 4`] = `
"<pre class="language-javascript">
  <code class="language-javascript">  <span class="token keyword">var</span> a <span class="token operator">=</span> <span class="token number">1</span><span class="token punctuation">;</span></code>
</pre>"
`;

exports[`code 5`] = `
"<pre class="language-none">
  <code class="language-none">  &lt;h1&gt;h1&lt;&#x2F;h1&gt;</code>
</pre>"
`;

exports[`code 6`] = `
"<div class="code-highlight">
  <table>
    <tbody>
      <tr>
        <td class="code-highlight-line"><pre class="language-none">
  <code class="language-none">
    <span>1</span>

  </code>
</pre></td>
        <td class="code-highlight-code"><pre class="language-none">
  <code class="language-none">  &lt;h1&gt;h1&lt;&#x2F;h1&gt;</code>
</pre></td>
      </tr>
    </tbody>
  </table>
</div>"
`;

exports[`code 7`] = `
"<pre class="language-none">
  <code class="language-none">  &lt;h1&gt;h1&lt;&#x2F;h1&gt;</code>
</pre>"
`;

exports[`code 8`] = `
"<div class="code-highlight">
  <table>
    <tbody>
      <tr>
        <td class="code-highlight-line"><pre class="language-none">
  <code class="language-none">
    <span>1</span>

  </code>
</pre></td>
        <td class="code-highlight-code"><pre class="language-html">
  <code class="language-html">  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>h1</span><span class="token punctuation">></span></span>h1<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>h1</span><span class="token punctuation">></span></span></code>
</pre></td>
      </tr>
    </tbody>
  </table>
</div>"
`;

exports[`code 9`] = `
"<pre class="language-html">
  <code class="language-html">  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>h1</span><span class="token punctuation">></span></span>h1<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>h1</span><span class="token punctuation">></span></span></code>
</pre>"
`;
